docs/rules/no-async-actions.md

```diff
@@ -0,0 +1,50 @@
+# Do not use async actions
+## Rule `no-async-actions`
+
+
```

Comment:
I'm a bit unclear on why we should not use async actions. Can you explain that more and include in the doc here?

Reply:
Me too not very clear about the requirement, i am exploring , will add more info once ready

Reply:
Let's add at least a brief explanation of the reasoning in this doc.